### PR TITLE
[ENT-1083]: Write Serializer to Collect Learner and Enrollment data

### PR DESCRIPTION
**Pull request by saleem-latif**
**Description:** 
Write utility functions/serializers to collect learner and enrollment data that would be sent as part of the xAPI statement.

**JIRA:** [ENT-1083](https://openedx.atlassian.net/browse/ENT-1083)

**Dependencies:**  https://github.com/edx/edx-enterprise/pull/328

**Merge checklist:**

- [x] Check that the versions of the requirements in the `platform-master.in` file match edx-platform.
- [x] New requirements are in the right place (`base.in` if only used in enterprise; in the correct `platform-****.in` files if they're hosted in edx-platform)
- [x] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [x] Called `make static` for webpack bundling if any static content was updated.
- [x] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Commits are (reasonably) squashed
- [x] Translations are updated
- [x] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)
- [ ] edx-platform PR (be sure to include edx-platform requirements upgrades that were present in this PR)

### CODE REVIEW

File: integrated_channels/xapi/serializers.py [hunk 1]
```diff
@@ -4,3 +4,56 @@
...
+class CourseInfoSerializer(serializers.Serializer):
+    """
+    Serializer for course info.
+
+    Model: CourseOverview
+    """
+    course_id = serializers.CharField(source='id')
+    course_title = serializers.CharField(source='display_name')
+    course_description = serializers.CharField(source='short_description')
+    course_details_url = serializers.CharField(source='marketing_url')
```
Comment (brittneyexline) [inline integrated_channels/xapi/serializers.py]: would we ever want to send the marketing url, since that will point to the B2C landing pages?

Comment (saleem-latif) [reply integrated_channels/xapi/serializers.py]: According to my understanding, the course-details page is the same for both B2B and B2C scenarios. This data will be sent as an extension to X-API object field that contains course details and will only be used by enterprise customer trying to get more info about a course. I think marketing URL is better than enterprise landing page. Let know your thoughts?

**Comment by saleem-latif** [comment]
@brittneyexline , @georgebabey Please take another look

File: integrated_channels/xapi/serializers.py (hunk 1 above)
Comment (georgebabey) [reply integrated_channels/xapi/serializers.py]: I think the marketing page for now makes the most sense. Maybe someday we'll have b2b marketing pages in the portal, but for now this is the best place to get course details.

File: integrated_channels/xapi/serializers.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    course_effore = serializers.CharField(source='effort')
```
Comment (georgebabey) [inline integrated_channels/xapi/serializers.py]: typo: `effort`

File: tests/test_integrated_channels/test_xapi/test_serializers.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            start=now,
+            end=now + timedelta(weeks=3, days=4),
+        )
+        self.course_overview = mock.Mock(**self.course_overview_mock_data)
+
+        self.expected_data = {
+            'course_description': self.course_overview_mock_data['short_description'],
+            'course_title': self.course_overview_mock_data['display_name'],
+            'course_duration': self.course_overview_mock_data['end'] - self.course_overview_mock_data['start'],
+            'course_effore': self.course_overview_mock_data['effort'],
```
Comment (georgebabey) [inline tests/test_integrated_channels/test_xapi/test_serializers.py]: same as above, typo: `effort`

File: integrated_channels/xapi/serializers.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+        Returns:
+            (str): string containing UUID for enterprise customer's Identity Provider.
+        """
+        try:
+            enterprise_learner = EnterpriseCustomerUser.objects.get(user_id=obj.id)
+        except EnterpriseCustomerUser.DoesNotExist:
+            return
+
+        return enterprise_learner.enterprise_customer.identity_provider
```
Comment (brittneyexline) [inline integrated_channels/xapi/serializers.py]: is this meant to get the learner's id passed through SSO from the IDP, or the slug of the IDP? the field name suggests the former, but the implementation and the comment suggest the latter.

Comment (georgebabey) [reply integrated_channels/xapi/serializers.py]: That list comes from the enrollment data reports and the `enterprise_sso_uid` has always referred to the user id. See https://github.com/edx/edx-documentation/blob/dccdfb3ff9053967415c00c855091fd6b026d65b/en_us/enterprise_api/source/api_reference/reference.rst

Brittney is correct in that this shouldn't be pointing to the slug, but instead `enterprise_learner.get_remote_id()`

**Comment by saleem-latif** [comment]
@georgebabey @brittneyexline I have incorporated your feedback, please take another look.

**Pull request merged**